src/iscn/parsing.ts

```diff
@@ -0,0 +1,97 @@
+import BigNumber from 'bignumber.js';
+import { getLikeWalletAddress, isValidAddress, changeAddressPrefix } from './addressParsing';
+import { getLikeWalletByLikerId } from './likerIdsAddresses';
+import { ISCNRecordData } from '../types';
+
+async function getLikeWalletFromId(
+  id: string,
+  { LIKE_CO_API_ROOT = 'https://api.like.co' }: { LIKE_CO_API_ROOT?: string } = {},
+)
+: Promise < { likeWallet: string | void | null } > {
+  let likeWallet: string | void | null = null;
+  let likerId: string | void | null = null;
+  const res = id.match(/^https:\/\/like\.co\/([a-z0-9_-]{6,20})/);
+  if (res) {
+    [, likerId] = res;
+    likeWallet = await getLikeWalletByLikerId(likerId, { LIKE_CO_API_ROOT });
+  } else {
+    likeWallet = getLikeWalletAddress(id);
+  }
+  return { likeWallet };
+}
+
+export async function getStakeholderMapFromParsedIscnData(
+  iscnData: ISCNRecordData,
+  { totalLIKE = 1, defaultWallet }: { totalLIKE?: number, defaultWallet?: string } = {},
+)
+: Promise < Map < string, { LIKE: number } > > {
+  const LIKEMap = new Map();
+  const { stakeholders } = iscnData;
+  if (stakeholders?.length) {
+    const likeWallets = stakeholders.map((stakeholder:any) => {
+      const id: string = stakeholder.entity['@id'];
+      if (isValidAddress(id)) {
+        return changeAddressPrefix(id, 'like');
+      }
+      return null;
+    });
+    const weightMap = new Map();
+    likeWallets.forEach((element: any, i) => {
+      if (element) {
+        let weight = new BigNumber(stakeholders[i].rewardProportion);
+        if (weightMap.has(element)) {
+          weight = weightMap.get(element).weight.plus(weight);
+        }
+        weightMap.set(element, { weight });
+      }
+    });
+    const totalWeight = [...weightMap.values()]
+      .map(({ weight }) => weight)
+      .reduce((prev, curr) => prev.plus(curr), new BigNumber(0));
+    weightMap.forEach(({ weight }, address) => {
+      const LIKE = new BigNumber(weight.times(totalLIKE).div(totalWeight).toFixed(9, 1)).toNumber();
+      LIKEMap.set(address, { LIKE });
+    });
+  }
+  if (LIKEMap.size === 0) {
+    if (!defaultWallet) throw new Error('No valid stakeholders and default wallet is not set');
```

Comment:
Not sure if we should throw Error or return empty map instead

Reply:
It seems to me that it is more appropriate to make the return type `Map<string, { LIKE: number }>> | null` and return `null` in this case.
By doing so, the caller is forced to do null checking before using the map, ensuring the application logic is robust.

Reply:
or just mark defaultWallet as required?